### ISSUE TITLE
StackStorm HA in rancher PV/PVC/Storage Classes

### ISSUE THREAD

**Issue by real747-david** [StackStorm/stackstorm-ha#168]
I setup K8s by rancher, 1 master and 3 workers, now I am pending on volumes. could somebady to show me which  PV/PVC/Storage Classes I could ues, how do I config it. thanks.
![image](https://user-images.githubusercontent.com/75285715/104146014-27d7f980-5404-11eb-8c71-61652c68d188.png)

**Comment by real747-david** [comment]
Version 0.50.0

**Comment by arms11** [comment]
I believe you should check the 3rd party helm charts for instance `MongoDB` on Bitnami to get that information. Should be `standard` for `Minikube` or `gp2` for EKS. Never worked with Rancher, so not sure on that end.

**Comment by armab** [comment]
In terms of storage requirements, ReadWriteOnce would work.

As @arms11 mentioned, persistent volumes are required by backends which are provisioned via Helm chart dependencies

Configuring storage for K8s cluster is out of scope of this repository and up to your K8s configuration, but I would look at Rancher docs like https://rancher.com/docs/rancher/v2.x/en/cluster-admin/volumes-and-storage/provisioning-new-storage/

**Comment by real747-david** [comment]
Thanks for your information, I just use Longhorn to create Storage Class, and all volumes was created automatically.  but I only found storageClassName: nfs in etcd part Values.yaml file, and my storageClassName is not "nfs", how can that volume created, and I did not find storageClassName configuration for mongodb and rabbitmq, but all 9 PVCs were created. I wonder why and how !

**Comment by armab** [comment]
I guess this K8s configuration is in place: https://kubernetes.io/docs/concepts/storage/dynamic-provisioning/ which is an automatic way to create the persistent volumes in K8s based on PersistentVolumeClaim storage requests coming from the specific Helm chart configurations (MongoDB, RabbitMQ, etcd in our case).

Example:
https://github.com/bitnami/charts/blob/master/bitnami/mongodb/templates/standalone/pvc.yaml

**Comment by armab** [comment]
Closing an issue as from what I understand you figured it out.

While exploring the stackstorm, please report any bugs in Github or if you have any more questions Forum https://forum.stackstorm.com/.
Thanks!
